File: src/main/java/io/github/kingg22/api/vacunas/panama/persistence/entity/Distrito.java
package io.github.kingg22.api.vacunas.panama.persistence.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import java.util.LinkedHashSet;
import java.util.Set;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.Nationalized;

@Entity
@Table(name = "distritos")
@NoArgsConstructor
@Getter
@Setter
public class Distrito {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", columnDefinition = "tinyint not null")
    private Short id;

    @Size(max = 100)
    @NotNull
    @Nationalized
    @Column(name = "nombre", nullable = false, length = 100)
    private String nombre;

    @NotNull
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "provincia", nullable = false)
    private Provincia provincia;

    @OneToMany(mappedBy = "distrito")
    @JsonBackReference
    private Set<Direccion> direcciones = new LinkedHashSet<>();
}

File: src/main/kotlin/io/github/kingg22/api/vacunas/panama/modules/direccion/entity/Distrito.kt
package io.github.kingg22.api.vacunas.panama.modules.direccion.entity

import io.github.kingg22.api.vacunas.panama.modules.direccion.domain.DistritoModel
import io.mcarle.konvert.api.KonvertFrom
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.Id
import jakarta.persistence.Index
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size

@Entity
@Table(
    name = "distritos",
    indexes = [
        Index(name = "ix_distritos_nombre", columnList = "nombre"),
    ],
)
@KonvertFrom(DistritoModel::class)
class Distrito(
    @Id
    @Column(name = "id", nullable = false)
    var id: Short? = null,

    @all:NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "provincia", nullable = false)
    var provincia: Provincia,

    @all:Size(max = 100)
    @all:NotNull
    @Column(name = "nombre", nullable = false, length = 100)
    var nombre: String,
) {
    override fun toString(): String = Distrito::class.simpleName +
        "(id=$id, provincia=$provincia, nombre='$nombre')"

    companion object
}
